R/lcd.R

```diff
@@ -44,13 +48,18 @@
 #' # pass curl options
 #' lcd(station = "02413099999", year = 2002, verbose = TRUE)
 #' }
-lcd <- function(station, year, ...) {
+lcd <- function(station, year, col_types = NULL, ...) {
   assert(station, c("character", "numeric", "integer"))
   assert(year, c("character", "numeric", "integer"))
   assert_range(year, 1901:format(Sys.Date(), "%Y"))
 
   path <- lcd_get(station = station, year = year, ...)
-  tmp <- safe_read_csv(path)
+  
+  if (is.null(col_types)) {
+    col_types <- lcd_columns()
+  }
+  assert(col_types, c("character"))
```

Comment:
In addition to checking col_types is character, it seems like you can run into trouble if the input to `col_types` is not of the correct length. e.g, 

```r
# using letters vector, length 26
lcd(station = "74746003904", year = 2010, col_types = letters)
#> Error in safe_read_csv(path, col_types = col_types) :
#>   file ~/Library/Caches/R/noaa_lcd/2010_74746003904.csv malformed; delete file and try again
```

Reply:
I think for the purpose of this function, we can specify that a named vector is expected, then the length of the vector does not matter. If the column name is misspecified by the user, `fread()`  returns a useful error that the column name was not found.
```r
# expect error
x <- letters
lcd(station = "74746003904", year = 2010, col_types = x)

Error in lcd(station = "74746003904", year = 2010, col_types = x) : 
  col_types must be a named vector, see lcd_columns() for an example and expected names
```

```r
# expect a useful error
names(x) <- x
lcd(station = "74746003904", year = 2010, col_types = x)

Error in lcd(station = "74746003904", year = 2010, col_types = x) : 
  a must equal a valid R class ('character', 'integer', 'numeric', 'factor', 'integer64', 'POSIXct')
b must equal a valid R class ('character', 'integer', 'numeric', 'factor', 'integer64', 'POSIXct')
c must equal a valid R class ('character', 'integer', 'numeric', 'factor', 'integer64', 'POSIXct')
d must equal a valid R class ('character', 'integer', 'numeric', 'factor', 'integer64', 'POSIXct')
```

```r
# error for wrong column name
lcd(station = "74746003904", year = 2010, col_types = c(station = "character"))

 Error in safe_read_csv(path, col_types = col_types) : 
  Column name 'station' (colClasses[[1]][1]) not found 
```

Reply:
sounds good, thanks